#!/usr/bin/env python
import os
import os.path as _path
import subprocess

try:
    script_dir = os.path.dirname(os.path.realpath(__file__))
    root = _path.normpath(_path.join(script_dir, '../../test'))

    print('')
    print('**** Running tests in python3 ****')
    subprocess.check_call(['python3', '-m', 'unittest', 'discover', root])

    print('Unit tests pass')
except subprocess.CalledProcessError as e:
    print('Error: Unit tests fail')
    raise e
